src/Storageless/Http/ClientFingerprint/Source.php

```diff
@@ -0,0 +1,13 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PSR7Sessions\Storageless\Http\ClientFingerprint;
+
+use Psr\Http\Message\ServerRequestInterface;
+
+interface Source
+{
+    /** @return non-empty-string */
+    public function extractFrom(ServerRequestInterface $request): string;
```

Comment:
This interface requires some documentation: taken as-is, it is not very clear what it does.

Also, could it be that it **cannot** extract information from `$request`?